docs/internal/DistributedArchitectureGuide.md

```diff
@@ -12,6 +12,218 @@ A guide to the general Elasticsearch components can be found [here](https://gith
 
 # Networking
 
+Every elasticsearch node maintains various networking clients and servers,
+protocols, and synchronous/asynchronous handling.
+
+## HTTP Transport
+The HTTP Transport Server (simply HTTP Transport) is a single entry point for
+all external clients (excluding cross-cluster communication). Management,
+ingestion, search, and all other external operations pass through the HTTP
+server.
+
+Elasticsearch works over HTTP 1.1 and supports features such as TLS, chunked
+transfer encoding, content compression, and pipelining. While attempting to
+be HTTP spec compliant, Elasticsearch is not a webserver. ES Supports `GET`
+requests with a payload (though some old proxies may drop content) and
+`POST` for clients unable to send `GET-with-body`. Requests cannot be cached
+by middle boxes.
+
+There is no connection limit, but a limit on payload size exists. The default
+maximum payload is 100MB after compression. It's a very large number and almost
+never a good target that the client should approach.
+
+Security features, including basic security (authc/authz/TLS) in the free tier,
+are achieved with separate x-pack modules.
+
+HTTP transport provides two options for content processing: aggregate fully and
+incremental. Aggregated content is a preferable choice for small messages that
+do not fit for incremental parsing (e.g., JSON). Aggregation has drawbacks, it
+requires more memory, which is reserved until all bytes are received. Concurrent
+incomplete requests can lead to unbounded memory growth and potential OOMs.
+Large delimited content, such as bulk indexing, which is processed in byte
+chunks. It's more complicated for application code, but provides better control
+over memory usage.
+
+Incremental bulk indexing includes a back-pressure feature.When memory pressure
+grows high, reading bytes from TCP sockets is paused for some connections,
+allowing only a few to proceed until the pressure is resolved. This mechanism
+protects against unbounded memory usage and OutOfMemory errors (OOMs).
+
+ES supports multiple `Content-Types` for the payload, collectively referred to
+internally as `XContentType`: CBOR, JSON, SMILE, YAML, and their versioned types.
```

Comment:
We can also handle other content-types, it's just that these are the best-supported ones.